aptos-move/framework/aptos-framework/sources/TokenV1.move

```diff
@@ -0,0 +1,935 @@
+        royalty: Royalty,
+        // The name of this Token
+        name: String,
+        // Describes this Token
+        description: String,
+        // the properties can be used to hold any custom attributes
+        default_property_list: PropertyList,
+        mutability_config: TokenMutabilityConfig,
+    }
+
+    /// Represents token resources owned by token owner
+    struct TokenStore has key {
+        tokens: Table<TokenId, Token>,
+        token_data: Table<TokenId, PropertyList>,
+        deposit_events: EventHandle<DepositEvent>,
+        withdraw_events: EventHandle<WithdrawEvent>,
+    }
+
+    /// Set of data sent to the event stream during a receive
+    struct DepositEvent has drop, store {
+        id: TokenId,
+        amount: u64,
+    }
+
+    /// Set of data sent to the event stream during a withdrawal
+    struct WithdrawEvent has drop, store {
+        id: TokenId,
+        amount: u64,
+    }
+
+    /// token creation event id of token created
+    struct CreateTokenEvent has copy, store, drop {
+        id: TokenId,
+        initial_balance: u64,
+    }
+
+    /// mint token event. This event triggered when creator adds more supply to existing token
+    struct MintTokenEvent has drop, store {
+        id: TokenId,
+        amount: u64,
+    }
+
+    //
+    // Core data structures for creating and maintaining tokens
+    //
+
+    /// Represent collection and token metadata for a creator
+    struct Collections has key {
+        collections: Table<String, Collection>,
+        token_data: Table<TokenDataId, TokenData>,
+        burn_capabilities: Table<TokenId, BurnCapability>,
+        mint_capabilities: Table<TokenId, MintCapability>,
+        create_collection_events: EventHandle<CreateCollectionEvent>,
+        create_token_events: EventHandle<CreateTokenEvent>,
+        mint_token_events: EventHandle<MintTokenEvent>,
+    }
+
+    /// Represent the collection metadata
+    struct Collection has store {
+        // Describes the collection
+        description: String,
+        // Unique name within this creators account for this collection
+        name: String,
+        // URL for additional information /media
+        uri: String,
+        // Total number of distinct Tokens tracked by the collection
+        count: u64,
+        // Optional maximum number of tokens allowed within this collections
+        maximum: u64,
+        mutability_config: CollectionMutabilityConfig,
+    }
+
+    /// create collection event with creator address and collection name
+    struct CreateCollectionEvent has drop, store {
+        creator: address,
+        collection_name: String,
+        uri: String,
+        description: String,
+        maximum:u64,
+    }
+
+    /// Capability required to mint tokens.
+    struct MintCapability has store {
+        token_id: TokenId,
+    }
+
+    /// Capability required to burn tokens.
+    struct BurnCapability has store {
+        token_id: TokenId,
+    }
```

Comment:
let's pause on these, I think we need to evaluate their goals and purposes. and also figure out our mental model on capabilities.